compiler/damlc/tests/src/daml-ghc-deterministic.sh

```diff
@@ -62,3 +63,32 @@ $protoc --decode_raw < "$TMP_OUT/out_proj_2" > "$TMP_OUT/decoded_out_proj_2"
 $diff -u "$TMP_OUT/decoded_out_proj_1" "$TMP_OUT/decoded_out_proj_2"
 $diff -u "$TMP_OUT/out_proj_1" "$TMP_OUT/out_proj_2"
 
+# Check that daml build is deterministic.
+# This includes things like the ZIP timestamps
+# in a DAR instead of just the package id.
+
+cat <<EOF > "$PROJDIR/daml.yaml"
+sdk-version: 0.0.0
+name: proj
+version: 0.0.1
+source: .
+dependencies: [daml-prim, daml-stdlib]
+EOF
+
+cat <<EOF > "$PROJDIR/A.daml"
+daml 1.2
+module A where
+EOF
+
+$damlc build --project-root "$PROJDIR" -o "$PROJDIR/out.dar"
+FIRST_SHA=$(sha256sum $PROJDIR/out.dar)
+
+$damlc build --project-root "$PROJDIR" -o "$PROJDIR/out.dar"
+SECOND_SHA=$(sha256sum $PROJDIR/out.dar)
```

Comment:
Does this test reliably fail without this change? It seems that ["Zip entry timestamps are recorded only to two 2 second precision"](https://www.mindprod.com/jgloss/zip.html). If not, maybe we could use something like `faketime` to make this test reliably fail if the timestamp is not deterministic?

Reply:
It sadly doesn’t fail reliably on a single-run but if you run it a couple of times, it fails reliably. I took a brief look at `libfaketime` last week but it didn’t seem worth the complexity for now.

Reply:
You could add a `sleep 10` in between the first and second build to make it fail more reliably.

(Plus, any code with `sleep` must be the good code.)

Reply:
This is a terrible idea, I love it :heart:

Reply:
> It sadly doesn’t fail reliably on a single-run but if you run it a couple of times, it fails reliably. I took a brief look at `libfaketime` last week but it didn’t seem worth the complexity for now.

That's fair.

> You could add a sleep 10 in between the first and second build to make it fail more reliably.

Yeah, let's try not to make CI intentionally slower.